src/main/java/org/betonquest/betonquest/quest/event/lever/LeverEvent.java

```diff
@@ -43,8 +46,25 @@ public void execute(final Profile profile) throws QuestRuntimeException {
                     + block.getZ() + " in world '" + block.getWorld().getName() + "'");
         }
 
-        final Powerable lever = (Powerable) block.getBlockData();
+        final Switch lever = (Switch) block.getBlockData();
         lever.setPowered(stateType.apply(lever.isPowered()));
         block.setBlockData(lever);
+        updateBlocksAround(block, lever);
+    }
+
+    private void updateBlocksAround(final Block block, final Switch powerableSwitch) {
+        final BlockFace attachedTo = switch (powerableSwitch.getAttachedFace()) {
+            case FLOOR -> BlockFace.DOWN;
+            case CEILING -> BlockFace.UP;
+            default -> powerableSwitch.getFacing().getOppositeFace();
+        };
+        final Block relative = block.getRelative(attachedTo);
+
+        final BlockState relativeState = relative.getState();
+        if (relativeState instanceof final Container container) {
+            container.getInventory().clear();
+        }
+        relative.setType(Material.AIR, false);
```

Comment:
Why is the block set to air?

Reply:
Its all an hack, to update the block. The set `relativeState.update(true);` restores everything.